vernac/g_vernac.mlg

```diff
@@ -36,6 +36,19 @@ open Attributes
 let vernac_kw = [ ";"; ","; ">->"; ":<"; "<:"; "where"; "at" ]
 let _ = List.iter CLexer.add_keyword vernac_kw
 
+(* For future use *)
+let reserved_non_keywords =
```

Comment:
How do you ensure that this list is going to be kept in sync with the code? It's already complicated enough to ensure that the doc gets updated to reflect changes to the code. This list already appears in the doc. Thus, I wouldn't insert this code until it is actually used.

Reply:
If we agree on the idea, this list can be automatically generated by `coqpp`. The naive way would be to have a `coqpp` token, say, `CMDIDENT` or whatever which `coqpp` would add to a list of reserved identifiers.

In the meantime, I removed it.

Reply:
And maybe automatically generated and inserted in the doc by doc_grammar.  But not any time soon.

Reply:
More on the way to implement a `CMDIDENT`: no need to change `coq_pp`. It is enough to work at the level of the Coq lexer (`tok.ml` and `cLexer.ml`), typically by adding `PCMDIDENT of string` and `CMDIDENT of string` in types `Tok.p` and `Tok.t`, then by instrumenting `tok_using` in `cLexer.ml` to add any `PCMDIDENT s` to the table of reserved identifiers. This is very easy.

I wonder though whether we could not be more ambitious and recognize the first identifier in _all_ rules and sons of these rules added to the `vernac` entry. By doing so, we would have an automatic way to recognize the first word of any command w/o doing any thing.

That would be a bit more complicated to implement though. We would first need a table of all entries that possibly start a vernac command. Then, we would need to hack `Pcoq.grammar_extend` to recognize when a "possibly-starting-sentence" entry is extended with some rule. All initial subentries of this rule should be added to the table of possibly-starting-sentence" entries. All initial ident of this rule are what we are looking for and have to registered to the table of reserved identifiers.

With this second approach we could also make a difference between say `Theorem` (in subentry `gallina`), `Print` (in subentry `command`), or `intros` (in subentry `simple_tactic`), ...

Do you have any preference?